deps/src/main/java/com/microsoft/azure/sdk/iot/deps/auth/IotHubSSLContext.java

```diff
@@ -319,9 +319,8 @@ private static Key parsePrivateKey(String privateKeyString) throws CertificateEx
             Security.addProvider(new BouncyCastleProvider());
 
             CertificateFactory certFactory = CertificateFactory.getInstance("X.509");
-            final PemReader publicKeyCertificateReader = new PemReader(new StringReader(publicKeyCertificateString));
 
-            try
+            try (PemReader publicKeyCertificateReader = new PemReader(new StringReader(publicKeyCertificateString)))
```

Comment:
Does the close happen automatically due to this new syntax?

Reply:
Yep, this is kind of like a using block in .NET

Reply:
Interesting